File: quickstart/kotlin/src/main/kotlin/Hello.kt
// Cryptohopper Kotlin SDK — hello world.
//
// Authenticates with a bearer token from $CRYPTOHOPPER_TOKEN, then makes
// two calls: ch.user.get() and ch.exchange.ticker(...).
//
// Run:
//   export CRYPTOHOPPER_TOKEN=your-40-char-bearer
//   ./gradlew run

import com.cryptohopper.Client
import com.cryptohopper.CryptohopperError
import kotlinx.coroutines.runBlocking
import kotlinx.serialization.json.contentOrNull
import kotlinx.serialization.json.jsonObject
import kotlinx.serialization.json.jsonPrimitive
import kotlin.system.exitProcess

fun main() = runBlocking {
    val token = System.getenv("CRYPTOHOPPER_TOKEN")
    if (token.isNullOrEmpty()) {
        System.err.println("Set CRYPTOHOPPER_TOKEN to a 40-char OAuth bearer first.")
        exitProcess(1)
    }

    val ch = Client.create(token)

    try {
        val me = ch.user.get()?.jsonObject
        val identity = me?.get("email")?.jsonPrimitive?.contentOrNull
            ?: me?.get("username")?.jsonPrimitive?.contentOrNull
            ?: me?.get("id")?.jsonPrimitive?.contentOrNull
            ?: "?"
        println("Logged in as $identity")

        val ticker = ch.exchange.ticker(exchange = "binance", market = "BTC/USDT")?.jsonObject
        println("BTC/USDT last: ${ticker?.get("last")?.jsonPrimitive?.contentOrNull}")
    } catch (e: CryptohopperError) {
        System.err.println("API error [${e.code}]: ${e.message}")
        e.ipAddress?.let { System.err.println("  Server saw your IP as: $it") }
        exitProcess(2)
    }
}

File: quickstart/php/hello.php
<?php
// Cryptohopper PHP SDK — hello world.
//
// Authenticates with a bearer token from $CRYPTOHOPPER_TOKEN, then makes
// two calls: user->get() and exchange->ticker().
//
// Run:
//   export CRYPTOHOPPER_TOKEN=your-40-char-bearer
//   composer install
//   php hello.php

declare(strict_types=1);

require __DIR__ . '/vendor/autoload.php';

use Cryptohopper\Sdk\Client;
use Cryptohopper\Sdk\CryptohopperException;

$token = getenv('CRYPTOHOPPER_TOKEN') ?: '';
if ($token === '') {
    fwrite(STDERR, "Set CRYPTOHOPPER_TOKEN to a 40-char OAuth bearer first.\n");
    exit(1);
}

$ch = new Client(apiKey: $token);

try {
    $me = $ch->user->get();
    $identity = $me['email'] ?? $me['username'] ?? $me['id'] ?? '?';
    echo "Logged in as {$identity}\n";

    $ticker = $ch->exchange->ticker(exchange: 'binance', market: 'BTC/USDT');
    echo "BTC/USDT last: {$ticker['last']}\n";
} catch (CryptohopperException $e) {
    fwrite(STDERR, "API error [{$e->getErrorCode()}]: {$e->getMessage()}\n");
    if ($e->getIpAddress()) {
        fwrite(STDERR, "  Server saw your IP as: {$e->getIpAddress()}\n");
    }
    exit(2);
}
